### PR TITLE
chore: remove react-bootstrap package

### LINKED ISSUE #1211: [imporvement] - consider to remove react-bootstrap package
### What bug did you find in the codebase?

currently the project uses react-bootstrap package for styles.
as far as i can see - there is no big need for `bootstrap` styles in the project...
(only the modal actually makes use of bootstrap package - but it can be implemented in other ways )
it can be nice to move to tailwind styling as it in the main project

### PR DESCRIPTION
**Pull request by shootermv**
## Summary of changes
removed `react-bootstrap` package
 `Modal` component replaced by native html modal element

## Checklist

- [x] I have read and followed the [contribution guidelines](https://github.com/freeCodeCamp/Developer_Quiz_Site/blob/main/CONTRIBUTING.md).
- [x] I have read through the [Code of Conduct](https://github.com/freeCodeCamp/Developer_Quiz_Site/blob/main/CODE_OF_CONDUCT.md) and agree to abide by the rules.
- [x] This PR is for one of the available issues and is not a PR for an issue already assigned to someone else.
- [x] My PR title has a short descriptive name so the maintainers can get an idea of what the PR is about.
- [x] I have provided a summary of my changes.

closes #1211

### CODE REVIEW

File: src/components/QuizModal.tsx [hunk 1]
```diff
@@ -18,56 +18,54 @@ const QuizModal: React.FC<QuizQuestion> = QuizQuestion => {
...
-        <Modal.Body>
-          <p>
-            <b>Your Answer:</b>
-          </p>
-          <p>{QuizQuestion.chosenAnswer}</p>
-          <p>
-            <b>Answer:</b>
-          </p>
-          <p>{QuizQuestion.displayExplanation}</p>
+    <Modal show={QuizQuestion.show}>
```
Comment (jdwilkin4) [inline src/components/QuizModal.tsx:21]: For the actual markup here, we don't need the Modal.Header, Modal.Title, etc stuff. 
That was something specific to React bootstrap
https://react-bootstrap.netlify.app/docs/components/modal/

But also, we don't need to recreate it either.
We can use the dialog element like this and use vanilla CSS for the styles in a `QuizModal.css` file

```
import React, { MouseEventHandler } from "react";
import "../stylesheets/QuizModal.css";

interface QuizQuestion {
  chosenAnswer: string;
  message: string;
  points: number;
  correct: boolean;
  displayExplanation: string;
  showReference: string;
  nextQuestion: MouseEventHandler;
  show: boolean;
}

const QuizModal: React.FC<QuizQuestion> = QuizQuestion => {
  const lightBulbUnicode = "\uD83D\uDCA1";
  const sadFaceUnicode = "\uD83D\uDE14";
  const answerIcon = QuizQuestion.correct ? lightBulbUnicode : sadFaceUnicode;
  return (
    <dialog open={QuizQuestion.show}>
      {/*TODO: Add appropriate classes and CSS to make the modal look like the one in production. */}
      <h2 className="modal-text">
        {answerIcon} {QuizQuestion.message}
      </h2>
      <h3 className="modal-text" data-testid="modal-points">
        Points: {QuizQuestion.points}
      </h3>

      <p>
        <b>Your Answer:</b>
      </p>
      <p>{QuizQuestion.chosenAnswer}</p>
      <p>
        <b>Answer:</b>
      </p>
      <p>{QuizQuestion.displayExplanation}</p>
      <a
        className="modal-link"
        href={QuizQuestion.showReference}
        target="_blank"
        rel="noopener noreferrer"
      >
        Learn more with this helpful resource
      </a>
      <br />
      <br />
      <p>
        Wanna learn how to code? <br /> Download the free{" "}
        <a
          className="modal-link"
          target="_blank"
          rel="noopener noreferrer"
          href="https://www.freecodecamp.org/news/learn-to-code-rpg/"
        >
          Learn to Code RPG game
        </a>
      </p>

      <button className="modal-btn" onClick={QuizQuestion.nextQuestion}>
        Next Question
      </button>
    </dialog>
  );
};
export default QuizModal;
```
We can also move the existing modal styles from the `App.css` file into the new `QuizModal.css` file

File: src/stylesheets/Modal.css [hunk 2]
```diff
@@ -0,0 +1,55 @@
+#modal {
```
Comment (jdwilkin4) [inline src/stylesheets/Modal.css:1]: This can be deleted too

**Review by jdwilkin4 — COMMENTED** [review]
Overall, we should go with the native HTML dialog element.
It has good support with browsers and a lot of built in functionality and accessibility features as well.

**Comment by shootermv** [comment]
agree, the native dialog is much more elegant solution!
will change pr asap

**Comment by jdwilkin4** [comment]
It looks like there are some conflicts that need to be resolved

**Pull request merged**